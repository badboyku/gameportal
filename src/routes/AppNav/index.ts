/* istanbul ignore file */
export { default as AppNav } from './AppNav';
