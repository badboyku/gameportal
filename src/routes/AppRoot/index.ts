/* istanbul ignore file */
export { default as AppRoot } from './AppRoot';
