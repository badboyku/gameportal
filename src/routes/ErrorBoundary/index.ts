/* istanbul ignore file */
export { default as ErrorBoundary } from './ErrorBoundary';
