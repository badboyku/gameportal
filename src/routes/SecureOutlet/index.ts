/* istanbul ignore file */
export { default as SecureOutlet } from './SecureOutlet';
