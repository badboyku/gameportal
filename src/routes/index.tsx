import { Outlet, RouteObject } from 'react-router-dom';
import { Home } from '../pages';

const Root = () => {
  return <Outlet />;
};

const routes: RouteObject[] = [
  {
    path: '/',
    element: <Root />,
    children: [{ path: '', element: <Home /> }],
  },
];

export default routes;
