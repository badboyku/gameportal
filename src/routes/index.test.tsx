import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import routes from './index';

jest.mock('../pages/Home', () => ({ Home: () => <div data-testid="home" /> }));

const renderRoutes = (initialEntries = ['/']) => {
  const router = createMemoryRouter(routes, { initialEntries });

  return render(<RouterProvider router={router} />);
};

describe('Component Routes', () => {
  it('renders without crashing', () => {
    const { asFragment } = renderRoutes();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the default route "Routes" -> "HomePage"', () => {
    renderRoutes();

    expect(screen.getByTestId('home')).toBeInTheDocument();
  });
});
