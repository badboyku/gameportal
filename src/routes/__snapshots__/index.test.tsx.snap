// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Routes renders without crashing 1`] = `
<DocumentFragment>
  <div
    data-testid="home"
  />
</DocumentFragment>
`;
